### PR TITLE
Support Kotlin 2.1.0-RC2

### PR DESCRIPTION
**Pull request by Mr3zee**
**Subsystem**
Compiler Plugin

**Problem Description**
Kotlin `2.1.0-RC` is out with breaking changes in Compiler API

**Solution**
Update and fix changes

### CODE REVIEW

File: versions-root/libs.versions.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [versions]
 # core library version
-kotlinx-rpc = "0.4.0"
+kotlinx-rpc = "0.5.0"
```
Comment (e5l) [inline versions-root/libs.versions.toml:3]: This would be nice to have in the separate commit.
Should we use `0.5.0-SNAPSHOT` instead?

File: gradle-conventions-settings/src/main/kotlin/conventions-repositories.settings.gradle.kts [hunk 2]
```diff
@@ -80,10 +95,27 @@ pluginManagement {
             mavenCentral()
             gradlePluginPortal()
         }
+
+        maven("${findGlobalRootDirPath()}/lib-kotlin/")
     }
 }
 
 gradle.rootProject {
+    fun findGlobalRootDirPath(): java.nio.file.Path {
```
Comment (e5l) [inline gradle-conventions-settings/src/main/kotlin/conventions-repositories.settings.gradle.kts:104]: could be lazy val

**Review by e5l — APPROVED** [review]

File: gradle-conventions-settings/src/main/kotlin/conventions-repositories.settings.gradle.kts (hunk 2 above)
Comment (Mr3zee) [reply gradle-conventions-settings/src/main/kotlin/conventions-repositories.settings.gradle.kts:104]: Would be the same, I instead used it once for invocation

File: versions-root/libs.versions.toml (hunk 1 above)
Comment (Mr3zee) [reply versions-root/libs.versions.toml]: Made as a separate commit
